[BUG] Nil pointer dereference in transfer-pvc progress tracking when rsync exhausts retries

**What version of crane are you running, and what are your clusters+platform**

- Crane: built from source (main branch)
- Source cluster: Kind (Kubernetes v1.35)
- Destination cluster: Kind (Kubernetes v1.35)
- Platform: Fedora 42, rootless Podman

**What did you expect to happen?**

When `transfer-pvc` rsync retries are exhausted, the command should report a clean failure with status "Failed".

**What actually happened?**

The process panics with a nil pointer dereference in `progress.go:304` because `p.TransferredData` is nil when rsync fails without transferring any data.

**Please include any relevant logs or errors**

```
Status: Preparing
Progress:
  Percentage:   <unavailable>
  Transferred:  <unavailable>
  Rate:         <unavailable>
  Retries:      5
Elapsed:        30s
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2b8d842]

goroutine 92 [running]:
github.com/konveyor/crane/cmd/transfer-pvc.(*Progress).Status(...)
    /home/rwurmbra/Desktop/crane/cmd/transfer-pvc/progress.go:304
github.com/konveyor/crane/cmd/transfer-pvc.(*Progress).AsString(0xc0005a7ae0)
    /home/rwurmbra/Desktop/crane/cmd/transfer-pvc/progress.go:237 +0x42
github.com/konveyor/crane/cmd/transfer-pvc.(*rsyncLogStream).Init.func1()
    /home/rwurmbra/Desktop/crane/cmd/transfer-pvc/progress.go:109 +0x3df
created by github.com/konveyor/crane/cmd/transfer-pvc.(*rsyncLogStream).Init in goroutine 1
    /home/rwurmbra/Desktop/crane/cmd/transfer-pvc/progress.go:79 +0x434
```

--- a/cmd/transfer-pvc/progress.go
+++ b/cmd/transfer-pvc/progress.go
@@ -306,7 +306,7 @@ func (p *Progress) Status() status {
 			return succeeded
 		}
 		if p.TransferredFiles == 0 &&
-			p.TransferredData.val == 0 &&
+			(p.TransferredData == nil || p.TransferredData.val == 0) &&
 			p.TotalFiles == nil {
 			return failed
 		}

--- a/cmd/transfer-pvc/progress_test.go
+++ b/cmd/transfer-pvc/progress_test.go
@@ -319,3 +319,25 @@ func TestAsString_ErrorsDisplayed(t *testing.T) {
 		})
 	}
 }
+
+func TestStatus_NilTransferredData(t *testing.T) {
+	exitCode := int32(1)
+	p := Progress{
+		ExitCode:         &exitCode,
+		TransferredData:  nil,
+		TransferredFiles: 0,
+		TotalFiles:       nil,
+		startedAt:        time.Now(),
+	}
+
+	defer func() {
+		if r := recover(); r != nil {
+			t.Errorf("Status() panicked with nil TransferredData: %v", r)
+		}
+	}()
+
+	s := p.Status()
+	if s != failed {
+		t.Errorf("Status() = %q, want %q", s, failed)
+	}
+}

